src/Interactive/Features/Interactive/Core/InteractiveHost.RemoteService.cs

```diff
@@ -116,34 +121,56 @@ internal void Dispose(bool joinThreads)
                 // set _disposing so that we don't attempt restart the host anymore:
                 _disposing = true;
 
+                if (Interlocked.Exchange(ref _processExitHandling, ProcessExitHandled) == ProcessExitHooked)
+                {
+                    Process.Exited -= ProcessExitedHandler;
+                    _disposeSemaphore.Set();
+                }
+
+                using (var semaphore = _disposeSemaphore)
```

Comment:
>using (var semaphore = _disposeSemaphore) [](start = 16, length = 41)

Why do you add using? There is no need for that. just call semaphore.Dispose() after calling semaphore.Wait